vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/index/VXQueryIndexReader.java

```diff
@@ -74,27 +77,49 @@
     private String[] childLocalName = null;
     private IFrameFieldAppender appender;
     private boolean firstElement;
+    private List<Byte[]> indexSeq;
 
     public VXQueryIndexReader(IHyracksTaskContext context, String indexPath, List<Integer> childSeq,
-            IFrameFieldAppender appender) {
+            List<Byte[]> indexSeq, IFrameFieldAppender appender) {
         this.ctx = context;
+        this.indexSeq = indexSeq;
         this.indexName = indexPath;
         this.appender = appender;
         final DynamicContext dCtx = (DynamicContext) ctx.getJobletContext().getGlobalJobData();
         childSequenceTypes = new ArrayList<>();
         for (int typeCode : childSeq) {
             childSequenceTypes.add(dCtx.getStaticContext().lookupSequenceType(typeCode));
         }
+        TextType tt = TextType.INSTANCE;
+        if (!indexSeq.isEmpty()) {
```

Comment:
Why add something to childSequenceType? Can you not have a separate process for index that happens after childSequenceType processing?